### PR TITLE
Rename DatePart to DateTimePart

### PR DESCRIPTION
**Pull request by am357**
As `DatePart` type includes Time components too, rename the type to
reflect the current value range.  In addition rename the related
functions.

GitHub Issue: https://github.com/partiql/partiql-lang-kotlin/issues/409

*Issue #, if available:* 409

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lang/src/org/partiql/lang/eval/ExprValueExtensions.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/eval/ExprValueExtensions.kt:103]: May also want to change the name of this extension function to `dateTimePartValue()`?

File: lang/src/org/partiql/lang/eval/builtins/DateAddExprFunction.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/eval/builtins/DateAddExprFunction.kt:69]: Could change error message to say "invalid date **time** part for ... "

File: lang/src/org/partiql/lang/eval/builtins/DateAddExprFunction.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/eval/builtins/DateAddExprFunction.kt:80]: Variable could also be renamed to `dateTimePart` and similarly on L79

File: lang/src/org/partiql/lang/syntax/LexerConstants.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/syntax/LexerConstants.kt:26]: Variable could be renamed to `DATE_TIME_PART_KEYWORDS` for consistency

File: lang/test/org/partiql/lang/eval/builtins/DateAddExprFunctionTest.kt
Comment (alancai98) [inline lang/test/org/partiql/lang/eval/builtins/DateAddExprFunctionTest.kt:75]: Function could be renamed to `parametersForInvalidDateTimePart`

File: lang/test/org/partiql/lang/eval/builtins/ExtractEvaluationTest.kt
Comment (alancai98) [inline lang/test/org/partiql/lang/eval/builtins/ExtractEvaluationTest.kt:90]: Variable could be renamed to `dateTimePart` and same with the variable on L127

**Review by alancai98 — COMMENTED** [review]
Change looks good overall. Left some minor comments about changing `datePart`/"date part" to `dateTimePart`/"date time part" respectively for consistency with this change.

May want to also do a code search for other comments, functions, and variables that need renamed. E.g. https://github.com/partiql/partiql-lang-kotlin/blob/feb0976d5355e6c71bf4fce4711b9d521abf5aa3/lang/test/org/partiql/lang/eval/EvaluatingCompilerTests.kt#L1753

**Comment by abhikuhikar** [comment]
I think Alan had already figured out few places where the renaming needs to be done. But there are many more occurrences that might have missed. For example, the documentation - `BuiltinFunctions.md` also needs to be modified. I would do a simple word search for "datePart" on a project/directory level and also look at the comments and other usages (such as the one in the comment, rightly pointed out by Alan).
I know this is a tedious task.  Thanks for taking this up.

**Comment by am357** [comment]
Good call-outs @abhikuhikar @alancai98, will send a new commit!

**Comment by am357** [comment]
Changes the rest of occurrences, more details of search result here:
https://paste.debian.net/hidden/9168be70/

**Comment by abhikuhikar** [comment]
Changes LGTM but can you also rename instances of "datePart" and "date part" in `BuiltInFunctions.md` file?

**Comment by am357** [comment]
> Changes LGTM but can you also rename instances of "datePart" and "date part" in `BuiltInFunctions.md` file?

That makes sense, will add.

**Review by alancai98 — APPROVED** [review]
LGTM. Mind if we merge this PR after the upcoming branch merge to `main`? I have some concerns that the files modified here will cause conflicts.

**Review by abhikuhikar — APPROVED** [review]

**Pull request merged**
